### PR TITLE
Request for team review: Enabling directorate

### PR DESCRIPTION
**Pull request by jaysonmc**
Proposal for how best to organize an enabling directorate

Work related to issue #254, requested by director

### CODE REVIEW

File: _pages/en/building-enabling-directorate.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+### Problem Statement
+
+In [Team Topologies](https://itrevolution.com/book/team-topologies/) they write that organizations should organize themselves by nesting the four team definitions.
+
+- Value Stream Aligned Teams
+- Platform Teams
+- Enabling Teams
+- Complex Subsystem Teams
+
+In our particular case, to use it as an example, the IT Strategy team is an enabling team. We are under a director who considers the directorate to be an enabling team.  Therefore the IT Strategy team is an enabling team within an enabling team, and has other sibling teams that are also enabling teams. That being the case, how would one synchronize activities of the entire directorate? Say the director was to offer their services to a team elsewhere in the organization. How best to manage and coordinate the work between the series of sub-enabling teams? That is the question this document is intending to answer.
```
Comment (remyb2canada) [inline _pages/en/building-enabling-directorate.md]: Would be worthwhile to have a one-two liner on what is an enabling team. e.g.
"those which are more expertise-focused. They facilitate other teams to do their work and to gain capabilities and be able to do all the things [like] testing, monitoring, deployment, etc., without requiring experts in each of those areas"

Comment (remyb2canada) [inline _pages/en/building-enabling-directorate.md]: Clarify that the scope of this document is focused on one directorate: ITSM (or IT Service & Vendor Management).
Unless the scope is that of the IT Strategy team towards the rest of IITB?
Or is the scope that of the ITSM team towards the rest of IITB?

File: _pages/en/building-enabling-directorate.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+### Proposal
+
+#### Organizational
+
+##### Internal Interactions
+
+The teams may remain in their present functional structure for now, which aligns well with the governments expected structure for the sake of processes such as the Public Service Performance Management (PSPM). In our particular case, the teams will be as follows:
+
+- IT Strategy
```
Comment (remyb2canada) [inline _pages/en/building-enabling-directorate.md]: I think we should state that the IT Strategy team's enabling value is aligned to their mandate (e.g. de-risking the use of IT. For example, identifying waste and excessive work in progress within the overall IT value chain).

File: _pages/en/building-enabling-directorate.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+[TO-DO: Quote from Team Topologies regarding varying levels of team communication]
+
+The amount of collaboration between the enabling teams and the enabled teams should vary significantly during the time which the enabling team is paired with the enabled team. As such, collaboration tools that allow for the rapid creation and disintegration should be leveraged. An example of a viable tool for such a requirement would be Slack, where groups can be created and disposed of with relative ease.
+
+#### Process
+
+##### Teams Iterative Process
+
+- All teams within the directorate have synchronized sprints
```
Comment (remyb2canada) [inline _pages/en/building-enabling-directorate.md]: That seems prescriptive. Why not let different teams use their own cadence?

File: _pages/en/building-enabling-directorate.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+###### Tooling
+
+[TO-DO: Quote from Team Topologies regarding varying levels of team communication]
+
+The amount of collaboration between the enabling teams and the enabled teams should vary significantly during the time which the enabling team is paired with the enabled team. As such, collaboration tools that allow for the rapid creation and disintegration should be leveraged. An example of a viable tool for such a requirement would be Slack, where groups can be created and disposed of with relative ease.
+
+#### Process
```
Comment (remyb2canada) [inline _pages/en/building-enabling-directorate.md]: Am I reading this correctly that this is an intake and WIP management process?
I.e. each ITSM teams (IT strategy, s/w, h/w, and IT service) have part of their time associated to enabling other teams. That means they have limitted capacity to intake enablement requests.

During the teams iterative process, each team will have to determine if they have capacity to take on new requests, or if their existing requests need to be re-prioritize.

File: _pages/en/building-enabling-directorate.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+##### Teams Iterative Process
```
Comment (remyb2canada) [inline _pages/en/building-enabling-directorate.md]: I would add another item:
- If an enabling team has a particular task that has remained on the KANBAN board for too long (e.g. 2 months), that task goes back in the backlog and may be picked up by another enabling team

File: _pages/en/building-enabling-directorate.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+- All teams within the directorate have synchronized sprints
+- Representatives from each enabling team within the enabling directorate meet prior to the synchronized sprint planning for the director to set directorate-level priorities
+- During the synchronized sprint planning meetings, ensure directorate-level priorities are broken down into manageable tasks
+
+##### Engaging the Enabling Teams
+
+- Potential teams to be enabled should contact the enabling teams
+- An item is created on the kanban board of the enabling team
```
Comment (remyb2canada) [inline _pages/en/building-enabling-directorate.md]: Maybe worthwhile to have a single kanban board for the whole directorate. A single "enabling" kanban board so that WIP and backlog is visible to the director in one place and has an easy monitoring ability to identify work that has been "in progress" for too long.

This is similar to the lean model where an hypothesis is produced and the validation of that hypothesis is not expected to take months otherwise it may indicate the hypothesis is false.

Comment (gcharest) [reply _pages/en/building-enabling-directorate.md]: This may be hard to manage but we could attempt it. Roll up of the cards may be a better option (not sure if there are solutions for this already).

File: _pages/en/building-enabling-directorate.md (hunk 1 above)
Comment (gcharest) [reply _pages/en/building-enabling-directorate.md]: Agreed, or a definitions section might be useful. We need to keep in mind most people reading our content will not have a similar background. Just linking books will not cut it 😀 (A lot of people don't have time/interest to read work related books)

Comment (gcharest) [reply _pages/en/building-enabling-directorate.md]: My understanding is that IT Strategy is an enabling team, same as each other teams in ITSM. So that makes our directorate an enabling directorate (a team of enabling teams).

File: _pages/en/building-enabling-directorate.md (hunk 2 above)
Comment (gcharest) [reply _pages/en/building-enabling-directorate.md]: "And establishing strategies to address those issues"

File: _pages/en/building-enabling-directorate.md (hunk 3 above)
Comment (gcharest) [reply _pages/en/building-enabling-directorate.md]: Hmm, unless it's planning at the directorate level? Not sure about this here.

File: _pages/en/building-enabling-directorate.md (hunk 4 above)
Comment (gcharest) [reply _pages/en/building-enabling-directorate.md]: I think since we are enabling teams, we don't have as much "requests"? Like, we get the odd off the wall request to do X but usually our role is to help other teams like the Value Stream Aligned Teams, which deal with requests?

File: _pages/en/building-enabling-directorate.md (hunk 5 above)
Comment (gcharest) [reply _pages/en/building-enabling-directorate.md]: That should be discussed automatically at the sprint planning, every two weeks. If it hasn't been completed, it should probably even be discussed at the retrospective.

File: _pages/en/building-enabling-directorate.md (hunk 6 above)
Comment (gcharest) [reply _pages/en/building-enabling-directorate.md]: Only thing I would caution is about redundant work. Need to avoid creating multiple "boards" to track the same things twice or more.

File: _pages/en/building-enabling-directorate.md (hunk 1 above)
Comment (jaysonmc) [reply _pages/en/building-enabling-directorate.md]: I tried to make it general so other teams could leverage it as well. Though @gcharest was right ..

Unless the scope is that of the IT Strategy team towards the rest of IITB? Yes
Or is the scope that of the ITSM team towards the rest of IITB? Yes

SARA is an 'Enabling Team'
ITPSM is an 'Enabling Team'
The teams within ITPSM are 'Enabling Teams'

That's the power of the model in the book is that the definitions get recursively applied. 

But the focus on the article is how ITPSM will interact. And in that model I also attempt to describe how the teams will interact with each other, and other teams, as part of that model.

File: _pages/en/building-enabling-directorate.md (hunk 2 above)
Comment (jaysonmc) [reply _pages/en/building-enabling-directorate.md]: I seem to recall a discussion where we suggested if folks are just re-wording or clarifying they should just modify the PR themselves. 

Discussion needed when folks would like to change the meaning of the idea/concept :)

File: _pages/en/building-enabling-directorate.md (hunk 3 above)
Comment (jaysonmc) [reply _pages/en/building-enabling-directorate.md]: Because for the sake of coordinating work between teams, it would be best if they were all working on the same coordinated tasks at the same time. 

If work comes down from the director to the managers, if ones team sprint planning just happened, and the work was intended to take only one sprint, that means if the sprints are malaligned they will not be able to accomodate the work without having a mid-sprint planning session.

TLDR; it is to enable easier coordination between the teams under the same director.

File: _pages/en/building-enabling-directorate.md (hunk 4 above)
Comment (jaysonmc) [reply _pages/en/building-enabling-directorate.md]: Could be a value stream aligned team, could be another enabling team. Could be anyone who wants to work better.

The idea is requests come from two places, and the team must prioritize how they manage the work during their planning meetings. 

Work may come from the director, if the entire directorate is enabling another directorate, for example. 

Work may come from one manager to another manage within ITPSM. 

In either case, the team must prioritize the work accoringly. 

In the case where the work comes from the director, this is when the synchonrized sprints come into play, as it will be hard to unify and coordinate work between 3 teams if they all out of sync when they are working on the work handed down by the director.

File: _pages/en/building-enabling-directorate.md (hunk 5 above)
Comment (jaysonmc) [reply _pages/en/building-enabling-directorate.md]: @remyb2canada 

"picked up by another enabling team" 

hmm .. that would be tricky, I imagine. Possible, but not sure I'd vote in favour of such a process (not commonly, any who). 

Perhaps we could discuss how that mechanic could work... have trouble imaging it atm

File: _pages/en/building-enabling-directorate.md (hunk 6 above)
Comment (jaysonmc) [reply _pages/en/building-enabling-directorate.md]: This is the idea of the synchronized sprints. 

Director meets with representative of teams
Teams discuss who will do what
Teams all have their sprint planning
Teams prioritize the work that they have taken on as part of the meeting with the director
Entire directorate completes sprint
Repeat (what did we all get done, what remains to be done)

So perhaps some master board with large tasks you can link to? Or perhaps milestones, if we all have our own projects or some such thing? I see the use case attempting to be solved here ... just not sure what tools would fit this requirement ... it's a neat idea though. 'sub kanban boards' 

Granted, that is supposed to be the responsibility of the product owner. Ensure they understand the greater priorities and goals and ensure that they're not doing double work and so forth. So the nested kanban board approach sounds like another way of solving the same problem the meetings with the director is supposed to solve (coordinating work between the teams in the directorate)
